src/otlp.zig

```diff
@@ -0,0 +1,744 @@
+                    if (req.attempts_counter < self.max_retries) {
+                        self.enqueue(req.options, req.attempts_counter + 1) catch |e| {
+                            std.debug.print("OTLP transport: failed to re-queue: {}\n", .{e});
+                        };
+                    }
+                },
+                else => |s| {
+                    // Non-retryable error should be logged
+                    std.debug.print("OTLP transport: exp backoff response has a non-retryable status: {s}\n", .{@tagName(s)});
+                },
+            }
+        }
+    }
+
+    fn calculateDelay(base_delay_ms: u64, max_delay_ms: u64, attempt: u32) i64 {
+        // Exponential backoff with jitter: delay = min(max_delay, base * 2^attempt) + random(0, 10%)
+        const delay: i64 = @intCast(@min(
+            max_delay_ms,
+            base_delay_ms * std.math.pow(u64, 2, attempt),
+        ));
+        var prng = std.Random.DefaultPrng.init(@intCast(std.time.timestamp()));
+        const jitter = prng.random().intRangeAtMost(i64, 0, @intCast(@divTrunc(delay, 10)));
+        return delay + jitter;
+    }
+
+    fn cloneFetchOptions(self: *Self, opts: http.Client.FetchOptions) !http.Client.FetchOptions {
+        var cloned = opts;
+        cloned.location = .{ .url = try self.allocator.dupe(u8, opts.location.url) };
+        if (opts.payload) |payload| {
+            cloned.payload = try self.allocator.dupe(u8, payload);
+        }
+        cloned.extra_headers = try self.allocator.dupe(http.Header, opts.extra_headers);
+
+        return cloned;
+    }
+
+    fn freeFetchRequest(self: *Self, req: *const FetchRequest) void {
+        self.allocator.free(req.options.location.url);
+        if (req.options.payload) |payload| {
+            self.allocator.free(payload);
+        }
+        self.allocator.free(req.options.extra_headers);
+    }
+};
+
+// This test is here to allow compiling all the code paths of ExpBackoffRetry struct.
+// In itself, it does not provide much value other than ensuring that the code compiles,
+// and there is no memory leak.
+test "otlp HTTPClient send fails for missing server" {
+    const allocator = std.testing.allocator;
+    var config = try ConfigOptions.init(allocator);
+    defer config.deinit();
+
+    const client = try HTTPClient.init(allocator, config);
+    defer client.deinit();
+
+    const url = try config.httpUrlForSignal(.metrics, allocator);
+    defer allocator.free(url);
+
+    var payload = [_]u8{0} ** 1024;
+    const result = client.send(url, &payload);
+    try std.testing.expectError(std.posix.ConnectError.ConnectionRefused, result);
+}
+
+const pbmetrics = @import("opentelemetry/proto/metrics/v1.pb.zig");
+const pblogs = @import("opentelemetry/proto/logs/v1.pb.zig");
+const pbtrace = @import("opentelemetry/proto/trace/v1.pb.zig");
+
+/// Export the data to the OTLP endpoint using the options configured with ConfigOptions.
+pub fn Export(
```

Comment:
This is the intended entrypoint for the exporter to call into the OTLP transport.